lib/diagram.dx

```diff
@@ -13,7 +13,7 @@ data Geom =
 
 HtmlColor : Type = Fin 3 => Word8
 
-foreign "showHex" showHexFFI : Word8 -> {IO} (Nat & RawPtr)
+foreign "showHex" showHexFFI : Word8 -> {IO} (NatRep & RawPtr)
```

Comment:
Could we avoid using `NatRep` in FFI declarations? I think that the point of `NatRep` is to make it easy to swap the implementations around, but FFI is exactly the place where _you don't want that_. You want a loud error that points out that you have to modify the other side of the FFI boundary too!

Reply:
good point